scripts/common.sh

```diff
@@ -17,6 +17,50 @@ post_to_slack() {
         https://slack.com/api/chat.postMessage > /dev/null
 }
 
+# Posts a comment to a GitHub PR. Requires a GitHub token is available in $GITHUB_TOKEN.
+# Usage: post_github_pr_comment "Hi!" "https://api.github.com/repos/<org>/<repo>/issues/<pr-number>/comments"
+post_github_pr_comment() {
+    local pr_comment=$1
+    local pr_comment_api_url=$2
+    local pr_comment_body=$(printf '{ "body": "%s" }' "$pr_comment")
+
+    curl \
+        -X POST \
+        -H "Authorization: token ${GITHUB_TOKEN}" \
+        -d "$pr_comment_body" \
+        $pr_comment_api_url > /dev/null
+}
+
+git_sha() {
+    echo "$(git rev-parse HEAD)"
+}
+
+git_sha_short() {
+    echo "$(git rev-parse --short HEAD)"
+}
+
+# Returns the name of the metadata file we expect to exist locally before running Pulumi.
+origin_bucket_metadata_filepath() {
+    echo "./origin-bucket-metadata.json"
+}
+
+# pr_number_or_git_sha returns either the PR number of the current GitHub Actions run or
+# the Git SHA of the current branch, for purposes of naming the destination bucket and
+# bundled build assets.
+pr_number_or_git_sha() {
+    if [[ "$GITHUB_EVENT_NAME" == "pull_request" && ! -z "$GITHUB_EVENT_PATH" ]]; then
+        echo "pr-$(cat "$GITHUB_EVENT_PATH" | jq -r ".number")"
+    else
+        echo "push-${GITHUB_SHA:=$(git_sha_short)}"
+    fi
+}
+
+# Get the AWS SSM Parameter Store key for the specified commit SHA. Used for mapping a
+# commit to a previously created bucket.
+ssm_parameter_key_for_commit() {
+    echo "pulumi-docs-bucket-for-commit-$1"
```

Comment:
I think ideally, we prefix this key to avoid polluting the SSM root keyspace. Maybe something like `/docs/commits/pulumi-docs-bucket-for-commit-$1`? This will complicate some of the later logic when parsing this value, but seems worthwhile to me.

Reply:
Thanks, I appreciate that. I haven't used SSM before, so I wasn't sure whether there was a more idiomatic way to use these keys. I'll make this change.

Reply:
For reference: https://docs.aws.amazon.com/systems-manager/latest/userguide/sysman-paramstore-su-organize.html